### ISSUE TITLE
Rock layers modified in presence of other mods (Worley's caves, potentially others). Breaks surface rock generation, ore generation, etc.

### ISSUE THREAD

**Issue by Claycorp** [TerraFirmaCraft/TerraFirmaCraft#361]
Surface rocks don't seem to match the top layer anymore in many cases.
![](https://i.imgur.com/Ix0HAtw.png)

Diorite surface rock on a Quartzite First layer, Dacite Second layer, Granite Thrid layer.
There is no Diorite in the ground even 32 blocks out at any level.

**Comment by alcatrazEscapee** [comment]
Seed + Location, World folder, or reproduction steps please

**Comment by Claycorp** [comment]
Seed: -6585409906966526004
You can find the same issue at -248 150 256
Conglomerate grass with Diorite rocks and Conglomerate rocks.

**Comment by reddler97** [comment]
have found this bug in my world as well (don't want to link seed because I have other mods), but if it helps i was looking for the border of the 2 top layers and the misplaced rock was the rock from the layer that was adjacent to the other one

**Comment by alcatrazEscapee** [comment]
It would very much help if you can link a seed, because I've been completely unable to reproduce this - the seed and location Clay linked doesn't show anything wrong for me.

**Comment by alcatrazEscapee** [comment]
After some testing, I've been able to reproduce this in dev by adding Worley's Caves. Potentially other mods would affect this, as it's unknown how exactly Worley's caves is modifying the rock layers.

What I've found:
 - After generation (ChunkGenTFC#L222), the rock layers seem to be fine.
 - During population (ChunkGenTFC#populate), the rock layers are mutated somehow, filled with other rocks. Unknown where this is getting modified, will need more debugging to find out where.
